src/cache/core/types/common.ts

```diff
@@ -46,7 +46,7 @@ export interface ReadFieldFunction {
 }
 
 export type ToReferenceFunction = (
-  object: StoreObject,
+  object: StoreObject | string | Reference,
```

Comment:
```suggestion
  objOrIdOrRef: StoreObject | string | Reference,
```
Might as well rename the `object` parameter here too.